### ISSUE TITLE
compiler bug for duplicate annotation

### ISSUE THREAD

**Issue by sgalles** [ceylon/ceylon-compiler#2260]
When this (erroneous) code is compiled
```ceylon
import java.beans {
	transient__GETTER
}

class Foo(){
	transient__GETTER
	transient__GETTER
	shared Integer bar = 42;
}
```
`compiler bug: compiler bug: multiple occurances of non-sequenced annotation class java.beans::Transient$Proxy at unknown`

**Comment by FroMage** [comment]
So this is not detected by the typechecker because Java annotations don't extend `OptionalAnnotation` but `ConstrainedAnnotation`, which we can only do because we erase it to `java.lang.annotation.Annotation`. We can't do the same for `OptionalAnnotation` as that would require `SequencedAnnotation` to be a subtype of it. I can only special-case it in the typechecker.

**Comment by FroMage** [comment]
Or I reverse the check from `is optional` to `!is sequenced` (but that would also match for annotations that are neither optional nor sequenced in Ceylon, where we'd have another error for them not being applicable here). Or I introduce `Declaration.isCeylon()`, or `Declaration.isJava()`, or I use the existing `Declaration.getScopedBackends()` to figure out if it's a native JVM thing.

The last one seems like the easiest, though not sure if it's correct.

**Comment by FroMage** [comment]
Ah shit, we're meant to accept this:

```ceylon
    postConstruct__SETTER
    postConstruct__GETTER
    Integer x;
```
